module/zfs/zfs_fm.c

```diff
@@ -1444,6 +1444,60 @@ zfs_ereport_fini(void)
 	mutex_destroy(&recent_events_lock);
 }
 
+void
+zfs_ereport_snapshot_post(const char *subclass, spa_t *spa, const char *name)
+{
+	nvlist_t *aux;
+
+	aux = fm_nvlist_create(NULL);
+	nvlist_add_string(aux, "snapshot_name", name);
```

Comment:
We should add an `FM_EREPORT_PAYLOAD_ZFS_SNAPSHOT_NAME` define for this in `include/sys/fm/fs/zfs.h`.

```c
#define FM_EREPORT_PAYLOAD_ZFS_SNAPSHOT_NAME "snapshot_name"
```

Reply:
roger

Reply:
Ahh is that what those are for

but they are still just "static" strings in the zedlets? I guess I need to find where `FM_EREPORT_PAYLOAD_ZFS_POOL` becomes `"${ZEVENT_POOL}"` in the zedlets.

Reply:
Right, they're all just static strings and from what I can tell the convention was always just to define all the valid payload names in one place.  On the ZED side we just convert whatever strings we get from the kernel to environment variable key/value pairs.